apps/test/unit/code-studio/progressReduxTest.js

```diff
@@ -1076,16 +1077,112 @@ describe('progressReduxTest', () => {
         focusAreaStageIds: []
       };
 
-      let categories = categorizedLessons(state, false);
-      assert.equal(categories.length, 1);
-      assert.equal(categories[0].levels.length, 1);
-      assert.equal(categories[0].levels[0].length, 0);
+      let groups = groupedLessons(state, false);
+      assert.equal(groups.length, 1);
+      assert.equal(groups[0].levels.length, 1);
+      assert.equal(groups[0].levels[0].length, 0);
+
+      groups = groupedLessons(state, true);
+      assert.equal(groups.length, 1);
+      assert.equal(groups[0].levels.length, 1);
+      assert.equal(groups[0].levels[0].length, 1);
+      assert.equal(groups[0].levels[0][0]['bonus'], true);
+    });
+
+    it('returns lesson group if a lesson has both lesson group and flex category', () => {
```

Comment:
great tests! If we do end up having to undo this change for any reason, and we tried to do it by just changing the one line in progressRedux.js, then I would imagine some of these tests might break, since they presumably depend on the new behavior. if that's true, then I would imagine we'd want to just roll back the entire PR, rather than trying to fix a single line.